### ISSUE TITLE
Test failures with Python 3.3

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1008]
5 tests fail with Python 3.3. I use Docutils trunk and Sphinx trunk.

Output of tests in attachment.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1008
- Originally reported by: Arfrever Frehtes Taifersar Arahesis
- Originally created at: 2012-09-19T01:14:18.087

**Comment by shimizukawa** [comment]
_From Dmitry Shachnev on 2012-10-25 15:10:05+00:00_

This bug is also a big problem for Debian/Ubuntu as we are switching to Python 3.3 by default now.

**Comment by shimizukawa** [comment]
_From Dmitry Shachnev on 2012-10-26 13:22:41+00:00_

I've been trying to debug this, the only issue I can find the reason to fail is: [this check](https://github.com/sphinx-doc/testing/blob/5b9f431fc92fc76c01fcb3b86c02c37298f1c2fc/sphinx/util/__init__.py#L200 now) always returns `True` in Python 3.3 (because every module now has a `__loader__`), so the first line of regexp in `test_build_html.py` needs to be adjusted to allow `<string>` as a module name.

Other issues are (I've no clue what may cause them):
- autodoc attempts to import `sphinx.application.TemplateBridge` module;
- strange
  
  ```
  >>  assert '1 added, 3 changed, 1 removed' in '1 added, 4 changed, 1 removed'
  ```
  
  in `test_env.py`.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2012-10-30 01:11:07+00:00_

I confirmed your reported test error with:
- Python 3.3 (32bit) on Windows7 x64
- Python 3.3 (32bit) on Ubuntu 12.04 x64
- docutils trunk (0.10)
- tox 1.4.2
- virtualenv 1.8.2
- nose 1.2.1
- Sphinx rev eb3967d1e43b (default)

tested by:

```
$ tox -e py33
```

sometime caused test error but sometime passed. I'm confusing..

BTW, I fixed a few errors.
- `__loader__` problem (maybe fixed):
  [in my fork a1f50d969fe8](https://bitbucket.org/shimizukawa/sphinx-fix-testing-fork/changeset/a1f50d969fe827e287a19342c565f542cb895c31)
- assert '1 added, 3 changed, 1 removed':
  [in my fork 03cb1c0a1c57](https://bitbucket.org/shimizukawa/sphinx-fix-testing-fork/changeset/03cb1c0a1c57c2cea11a9b0e5af6b605dc2302e1)

I'll continue to fix.

**Comment by shimizukawa** [comment]
_From Dmitry Shachnev on 2012-10-30 09:33:59+00:00_

Thanks!

Note that some issues can be fixed by patch from issue #998.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2012-10-30 13:26:11+00:00_

 # I found error reason, but I don't know why working before Python-3.3...

Dmitry, please change your code and test again.

```
diff -r b1c22e958d7b sphinx/ext/autodoc.py
--- a/sphinx/ext/autodoc.py     Tue Oct 30 20:41:05 2012 +0900
+++ b/sphinx/ext/autodoc.py     Tue Oct 30 22:23:31 2012 +0900
@@ -1102,13 +1102,13 @@
 class MethodDocumenter(DocstringSignatureMixin, ClassLevelDocumenter):
     """
     Specialized Documenter subclass for methods (normal, static and class).
     """
     objtype = 'method'
     member_order = 50
-    priority = 0
+    priority = 1
```

**Comment by shimizukawa** [comment]
_From Dmitry Shachnev on 2012-10-30 16:48:24+00:00_

[Takayuki Shimizukawa](https://bitbucket.org/shimizukawa): After that change, **everything** works! Thanks very much!

Can you please commit it and propose a pull request?

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2012-10-31 04:31:24+00:00_

[Dmitry Shachnev](https://bitbucket.org/mitya57): Thanks for your confirmation!

I sent [PR83](https://bitbucket.org/birkenfeld/sphinx/pull-request/83/fix-python-33-incompatibility/diff).

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2012-11-01 16:54:57+00:00_

fix #1008 #1029 : intersphinx_mapping values are not stable if mapping have plural key/value set

→ <<cset d856e89a79768f572e627ca927d14ff51f6892cb>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2012-11-01 16:54:57+00:00_

fix #1008: can't get correct source code filename because all modules have '**loader**' attribute at Python-3.3.

→ <<cset c1145e65bac0c5fe912baea62f02ae4fbba67405>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2012-11-01 16:54:57+00:00_

fix #440 #1008 : coarse timestamp resolution in some filesystem generate wrong outdated file-list

→ <<cset 718993afc8e0e61c689a7681aa27f0f6dfbbcb7e>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2012-11-01 16:54:58+00:00_

fix #1008: fix autodoc documenter resolve ordering.

note: MethodDocumenter and FunctionDocumenter is same priority then
Documenter.format_signature() choice documenter is not stable
(depend on memory condition?).

→ <<cset 88e1bbe307de1270bcd1257769918c267a8d6d09>>
